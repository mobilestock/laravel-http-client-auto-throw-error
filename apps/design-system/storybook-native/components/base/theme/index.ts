import { Dimensions, PixelRatio, Platform, StatusBar } from 'react-native'

export const theme = {
  cores: {
    corSecundaria: '#4ba8ff',
    botaoPadrao: '#4B6680',
    backgroundGray: '#7D8D9C',
    tomate: '#FF6C63',

    azul100: '#4ba8ff',
    azul40: '#222465',
    azul30: '#014C76',
    azul20: '#96CCFF',
    azul: 'blue',

    branco90: '#f1f1f1',
    branco: '#fff',

    bege10: '#c4c4c4',

    preto: '#000',

    texto: '#292929',

    sucesso: '#0cb536',

    alerta80: '#F51D07',
    alerta80A_bb: '#F51D07bb',
    alerta70: '#FF3D0D',
    alerta50: '#E86513',
    alerta20: '#F2BF2C',

    whatsapp: '#128c7e',

    situacoes: {
      conferencia: {
        background: '#363636',
        color: '#eeeeee'
      },
      expedicao: {
        background: '#008DDC',
        color: '#f1f1f1'
      },
      pontoTransporte: {
        background: '#DC7700',
        color: '#f1f1f1'
      },
      entregarCliente: {
        background: '#00DC58',
        color: '#000'
      },
      entregue: {
        background: '#00DC58',
        color: '#f1f1f1'
      },
      devolucao: {
        background: '#8B71D9',
        color: '#000'
      },
      monitoramento: {
        color: '#FFF',
        background: '#4ba8ff'
      }
    }
  },
  fonts: {
    size(fontSize: number, standardScreenHeight = 680): number {
      const { height, width } = Dimensions.get('window')
      const standardLength = width > height ? width : height
      let offset = 0
      let heightPercent = 0

      if (width > height) {
        offset = 0
      } else if (Platform.OS === 'ios') {
        offset = 190
      } else {
        offset = StatusBar.currentHeight || 0
      }

      const deviceHeight = Platform.OS !== 'android' ? standardLength - offset : standardLength

      if (typeof fontSize === 'number') {
        heightPercent = (fontSize * deviceHeight) / standardScreenHeight
      } else {
        heightPercent = (parseFloat(fontSize) * deviceHeight) / 100
      }
      return Math.round(heightPercent)
    }
  },
  layout: {
    size: (size: number): number => {
      if (PixelRatio.get() < 2) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 1.3
      }
      if (PixelRatio.get() >= 2 && PixelRatio.get() < 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.9
      }
      if (PixelRatio.get() >= 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.7
      }
      return 16
    },
    width: (widthPercent: number): number => {
      let screenWidth = Dimensions.get('window').width
      const elemWidth = typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent)
      return PixelRatio.roundToNearestPixel((screenWidth * elemWidth) / 100)
    },
    height: (heightPercent: number): number => {
      let screenHeight = Dimensions.get('window').height
      const elemHeight = typeof heightPercent === 'number' ? heightPercent : parseFloat(heightPercent)
      return PixelRatio.roundToNearestPixel((screenHeight * elemHeight) / 100)
    }
  }
}
