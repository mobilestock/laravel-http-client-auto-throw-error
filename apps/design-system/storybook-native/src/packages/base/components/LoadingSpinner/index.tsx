import React from 'react'
import { ActivityIndicator, ViewProps } from 'react-native'
import styled from 'styled-components/native'

interface LoadingSpinnerProps extends ViewProps {
  title?: string
}
export const LoadingSpinner: React.FC<LoadingSpinnerProps> = props => {
  return (
    <Container {...props}>
      {props.children || (
        <>
          <Loading size="large" />
          <Text>{props.title}</Text>
        </>
      )}
    </Container>
  )
}
const Container = styled.View`
  justify-content: center;
  align-items: center;
  position: relative;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.colors.decorator.pure};
`
const Text = styled.Text``
const Loading = styled(ActivityIndicator)`
  color: ${({ theme }) => theme.colors.decorator.shadow};
`
