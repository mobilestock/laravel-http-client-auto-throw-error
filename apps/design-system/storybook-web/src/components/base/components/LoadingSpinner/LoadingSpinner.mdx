import { Canvas, Meta, Story } from '@storybook/blocks';
import * as LoadingSpinnerStories from './index.stories';

<Meta of={LoadingSpinnerStories} />

# LoadingSpinner

O componente **LoadingSpinner** é utilizado para demonstrar um estado de carregamento.

## Uso Básico

Você pode utilizar o LoadingSpinner para demonstrar um estado de carregamento.

<Canvas of={LoadingSpinnerStories.UsoBasico} />
