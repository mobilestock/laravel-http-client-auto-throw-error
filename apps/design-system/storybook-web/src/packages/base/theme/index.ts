export const theme = {
  colors: {
    background: '#f05'
  }
}
