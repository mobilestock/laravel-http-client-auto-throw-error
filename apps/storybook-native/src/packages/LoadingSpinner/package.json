{
  "name": "@mobilestock-native/loading-spinner",
  "version": "0.0.2",
  "main": "./src/index.tsx",
  "files": [
    "*"
  ],
  "exports": {
    ".": {
      "default": "./src/index.tsx"
    }
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "styled-components": "^6.1.8"
  }
}