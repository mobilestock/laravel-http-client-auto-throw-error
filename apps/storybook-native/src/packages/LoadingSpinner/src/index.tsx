import React from 'react'
import { ActivityIndicator, ViewProps } from 'react-native'
import styled, { useTheme } from 'styled-components/native'

interface LoadingSpinnerProps extends ViewProps {
  title?: string
}

const LoadingSpinner: React.FC<LoadingSpinnerProps> = (props) => {
  const theme = useTheme()
  return (
    <Container testID="loading-spinner-container" {...props}>
      {props.children || (
        <>
          <Loading testID="activity-indicator" size="large" color={theme.colors.container.shadow} />
          <Text>{props.title}</Text>
        </>
      )}
    </Container>
  )
}

const Container = styled.View`
  justify-content: center;
  align-items: center;
  position: relative;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.colors.container.default};
`
const Text = styled.Text``
const Loading = styled(ActivityIndicator)``

export default LoadingSpinner
