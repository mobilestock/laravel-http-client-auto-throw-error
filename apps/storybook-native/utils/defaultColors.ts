export default {
  primary: '#029afa',
  shadow: 'rgba(0 0 0 0.25)',
  white: '#FFFFFF',
  black: '#000000',
};
