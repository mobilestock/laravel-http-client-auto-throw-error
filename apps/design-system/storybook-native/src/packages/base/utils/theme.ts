import { Dimensions, PixelRatio, Platform, StatusBar } from 'react-native'

import defaultColors from './defaultColors'

export const theme = {
  colors: {
    text: {
      primary: defaultColors.black,
      secondary: defaultColors.white,
    },

    container: {
      shadow: defaultColors.shadow,
      pure: defaultColors.white,
    },

    button: {
      default: defaultColors.primary,
    },
  },

  layout: {
    size: (size: number): number => {
      if (PixelRatio.get() < 2) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 1.3
      }
      if (PixelRatio.get() >= 2 && PixelRatio.get() < 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.9
      }
      if (PixelRatio.get() >= 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.7
      }
      return 16
    },
    width: (widthPercent: number): number => {
      let screenWidth = Dimensions.get('window').width
      const elemWidth = typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent)
      return PixelRatio.roundToNearestPixel((screenWidth * elemWidth) / 100)
    },
    height: (heightPercent: number): number => {
      let screenHeight = Dimensions.get('window').height
      const elemHeight = typeof heightPercent === 'number' ? heightPercent : parseFloat(heightPercent)
      return PixelRatio.roundToNearestPixel((screenHeight * elemHeight) / 100)
    },
  },

  fonts: {
    size(fontSize: number, standardScreenHeight = 680): number {
      const { height, width } = Dimensions.get('window')
      const standardLength = width > height ? width : height
      let offset = 0
      let heightPercent = 0

      if (width > height) {
        offset = 0
      } else if (Platform.OS === 'ios') {
        offset = 190
      } else {
        offset = StatusBar.currentHeight || 0
      }

      const deviceHeight = Platform.OS !== 'android' ? standardLength - offset : standardLength

      if (typeof fontSize === 'number') {
        heightPercent = (fontSize * deviceHeight) / standardScreenHeight
      } else {
        heightPercent = (parseFloat(fontSize) * deviceHeight) / 100
      }
      return Math.round(heightPercent)
    },
  },
}
