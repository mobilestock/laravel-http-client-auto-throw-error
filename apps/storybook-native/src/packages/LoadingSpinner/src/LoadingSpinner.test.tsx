import { render } from '@testing-library/react-native'
import React from 'react'
import { Text } from 'react-native'
import { ThemeProvider } from 'styled-components/native'
import LoadingSpinner from '.'
import { theme } from '../../../../utils/theme'

function withProviders(ui: React.ReactElement) {
  return <ThemeProvider theme={theme}>{ui}</ThemeProvider>
}

describe('LoadingSpinner', () => {
  it('deve renderizar o ActivityIndicator e o título quando children não são fornecidos', () => {
    const title = 'Carregando...'
    const { getByText, getByTestId } = render(withProviders(<LoadingSpinner title={title} />))

    expect(getByText(title)).toBeTruthy()
    expect(getByTestId('activity-indicator')).toBeTruthy()
  })

  it('deve renderizar os children quando fornecidos', () => {
    const childText = 'Conteúdo personalizado'
    const { getByText, queryByTestId } = render(
      withProviders(
        <LoadingSpinner>
          <Text>{childText}</Text>
        </LoadingSpinner>,
      ),
    )

    expect(getByText(childText)).toBeTruthy()
    expect(queryByTestId('activity-indicator')).toBeNull()
  })

  it('deve passar as props para o Container', () => {
    const testID = 'loading-spinner-container'
    const { getByTestId } = render(withProviders(<LoadingSpinner testID={testID} />))

    const container = getByTestId(testID)
    expect(container).toBeTruthy()
  })

  it('deve aplicar o estilo correto ao ActivityIndicator', () => {
    const { getByTestId } = render(withProviders(<LoadingSpinner />))
    const activityIndicator = getByTestId('activity-indicator')

    expect(activityIndicator.props.color).toBe(theme.colors.container.shadow)
    expect(activityIndicator.props.size).toBe('large')
  })

  it('deve corresponder ao snapshot quando children não são fornecidos', () => {
    const { toJSON } = render(withProviders(<LoadingSpinner title="Carregando..." />))
    expect(toJSON()).toMatchSnapshot()
  })

  it('deve corresponder ao snapshot quando children são fornecidos', () => {
    const { toJSON } = render(
      withProviders(
        <LoadingSpinner>
          <Text>Conteúdo personalizado</Text>
        </LoadingSpinner>,
      ),
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
