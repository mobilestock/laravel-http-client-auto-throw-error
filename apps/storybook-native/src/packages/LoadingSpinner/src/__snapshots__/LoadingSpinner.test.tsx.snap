// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner deve corresponder ao snapshot quando children não são fornecidos 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "position": "relative",
      "top": 0,
      "width": "100%",
    }
  }
  testID="loading-spinner-container"
  title="Carregando..."
>
  <ActivityIndicator
    color="rgba(0 0 0 0.25)"
    size="large"
    style={{}}
    testID="activity-indicator"
  />
  <Text
    style={{}}
  >
    Carregando...
  </Text>
</View>
`;

exports[`LoadingSpinner deve corresponder ao snapshot quando children são fornecidos 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "position": "relative",
      "top": 0,
      "width": "100%",
    }
  }
  testID="loading-spinner-container"
>
  <Text>
    Conteúdo personalizado
  </Text>
</View>
`;
